### PR TITLE
✅ Basic Trading Integration Testing

### PR DESCRIPTION
**Pull request by JaredBorders**
closes https://github.com/Kwenta/margin-manager/issues/111
closes https://github.com/Kwenta/margin-manager/issues/112

### CODE REVIEW

File: src/MarginBase.sol [hunk 1]
```diff
@@ -329,54 +328,25 @@ contract MarginBase is MinimalProxyable, IMarginBase, OpsReady {
...
 
-    function _perpsV2Deposit(address _market, int256 _amount) internal {
-        if (_amount <= 0) {
-            revert InvalidMarginDelta();
-        } else if (_abs(_amount) > freeMargin()) {
-            revert InsufficientFreeMargin(freeMargin(), _abs(_amount));
-        } else {
+    function _perpsV2ModifyMargin(address _market, int256 _amount) internal {
+        if (_amount > 0) {
+            if (_abs(_amount) > freeMargin()) {
```
Comment (JChiaramonte7) [inline src/MarginBase.sol:333]: Technically don't need `_abs`

File: test/integration/account.behavior.t.sol [hunk 2]
```diff
@@ -204,16 +221,57 @@ contract AccountBehaviorTest is Test {
...
+                    IMarginBase.ValueCannotBeZero.selector,
+                    valueName
+                )
+            );
+
+            // attempt to deposit zero sUSD into account
+            account.deposit(x);
+        } else if (x > AMOUNT) {
+            // expect revert
+            vm.expectRevert();
```
Comment (JChiaramonte7) [inline test/integration/account.behavior.t.sol:241]: Comments like this prob not useful

File: src/MarginBase.sol (hunk 1 above)
Comment (JaredBorders) [reply src/MarginBase.sol:333]: freeMargin() returns a uint

Comment (JaredBorders) [reply src/MarginBase.sol:333]: I can just cast instead of calling abs tho

Comment (JaredBorders) [reply src/MarginBase.sol:333]: Your code used the same conv technique which is why I followed:
```
if (_marginDelta > 0) {
            // ensure margin doesn't exceed max
            if (_abs(_marginDelta) > freeMargin()) {
                revert InsufficientFreeMargin(freeMargin(), _abs(_marginDelta));
            }
            committedMargin += _abs(_marginDelta);
        }
```

File: test/integration/account.behavior.t.sol [hunk 3]
```diff
@@ -226,34 +284,168 @@ contract AccountBehaviorTest is Test {
...
+
+        // expect free margin to be equal to AMOUNT
+        assert(account.freeMargin() == AMOUNT);
+
+        // withdraw all sUSD from account
+        account.withdraw(AMOUNT);
+
+        // expect free margin to be zero
+        assert(account.freeMargin() == 0);
+    }
```
Comment (JChiaramonte7) [inline test/integration/account.behavior.t.sol:426]: This seems more fitting as a unit test

Comment (JaredBorders) [reply test/integration/account.behavior.t.sol:426]: I agree. Will move

**Pull request merged**